front/src/applications/editor/tools/rangeEdition/tool-factory.tsx

```diff
@@ -80,50 +82,72 @@ function getRangeEditionTool<T extends EditorRange>({
-              onClick({ setState }) {
-                setState(getInitialState());
-              },
-            }
-          : {
-              id: 'new-catenary',
-              icon: IoMdAddCircleOutline,
-              labelTranslationKey: 'Editor.tools.catenary-edition.actions.new-catenary',
-              onClick({ setState }) {
-                setState(getInitialState());
-              },
-            },
-        layersEntity.objType === 'SpeedSection'
-          ? {
-              id: 'reset-speed-section',
-              icon: BiReset,
-              labelTranslationKey: 'Editor.tools.speed-edition.actions.reset-speed-section',
-              isDisabled({ state: { entity, initialEntity } }) {
-                return isEqual(entity, initialEntity);
-              },
-              onClick({ setState, state: { initialEntity } }) {
-                setState({
-                  entity: cloneDeep(initialEntity),
-                });
-              },
-            }
-          : {
-              id: 'reset-catenary',
-              icon: BiReset,
-              labelTranslationKey: 'Editor.tools.catenary-edition.actions.reset-catenary',
-              isDisabled({ state: { entity, initialEntity } }) {
-                return isEqual(entity, initialEntity);
-              },
-              onClick({ setState, state: { initialEntity } }) {
-                setState({
-                  entity: cloneDeep(initialEntity),
-                });
-              },
-            },
+        {
+          id: `new-${getObjTypeAction(layersEntity.objType)}`,
+          icon: IoMdAddCircleOutline,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.new-${getObjTypeAction(layersEntity.objType)}`,
+          onClick({ setState }) {
+            setState(getInitialState());
+          },
+        },
+        {
+          id: `reset-${getObjTypeAction(layersEntity.objType)}`,
+          icon: BiReset,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.reset-${getObjTypeAction(layersEntity.objType)}`,
+          isDisabled({ state: { entity, initialEntity } }) {
+            return isEqual(entity, initialEntity);
+          },
+          onClick({ setState, state: { initialEntity } }) {
+            setState({
+              entity: cloneDeep(initialEntity),
+            });
+          },
+        },
+      ],
+      [
+        {
+          id: 'delete-entity',
+          icon: BsTrash,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.delete-${getObjTypeAction(layersEntity.objType)}`,
+          // Show button only if we are editing
+          isDisabled({ state }) {
+            return state.initialEntity.geometry.coordinates.length === 0;
```

Comment:
If I create a catenary and save, I can't delete it. I need to go back to the select panel, select it and then I can delete it. Why don't you check whether initialEntity has id or not instead of geometry.coordinates ?
```suggestion
            return state.initialEntity.properties.id === NEW_ENTITY_ID;
```

Reply:
Oh didn't see we had that const. Lost so much time finding a way for each objtype (which moreover didn't work).
Thank you for spotting it :)

Reply:
Actually it works inly for the rangeEdition form. As it's a custom one, we do a `initialEntity: cloneDeep(savedEntity),` when saving but it's not the case in others form.

Reply:
Finally, I changed the behavior in switchEdition and pointEdition to match this one. It's the best way to be able to delete the created object directly.